### ISSUE TITLE
Configuring PHPStorm for Platform

### ISSUE THREAD

**Issue by AngryWookie** [platformsh/legacy-cli#55]
Since I have started using Platform I've been running into an issue that has caused frustration and I wanted to get feedback from the Platform team, or anyone that has run into this, as to how we handle this issue. The issue has to do with the way Platform creates a new folder in the 'builds' directory then uses a symlink for the www directory to actually display the site. This is causing a number of issues in my IDE (PHPStorm) and I have yet to find a good way to resolve it. By having so many items in the builds folder it makes things like object search and jumping to a declaration a bit trickier because I end up getting presented with a dialog asking me which object or declaration I'd like to look at, this list continues to grow as more builds are created. This adds a bit of extra time and adds the possibility I might click on the wrong one and be looking at out of date information. The number of builds also increases index time which can sometimes bog down the IDE/machine.

I've tried marking the builds folder as excluded in my PHPStorm project but when I do that it makes debugging more complex. Each time a request is made to index.php it throws an error because that is now outside the scope of the project. I can configure PHPStorm to map the symlink properly but the next time I do a build I have to recreate that mapping in the IDE.

I'd just like to know if there are any suggestions on how to work around this. So far I've just accepted it but I'm afraid this is the sort of thing that might push people off when they try Platform.

**Comment by marcvangend** [comment]
I can't get this to work properly, it seems that every approach has one or more disadvantages.

The suggestion made by @DuaelFr will probably work for certain projects, but not for mine, because we have some 3rd party php libraries in www.

The method of @pjcdawkins has the disadvantage that PhpStorm keeps popping up a dialog when I want to navigate to a function delaration (Ctrl-click or Ctrl-B) in a custom module. You can solve that by marking the respository directory as excluded as well, but then you don't get the built in Git status and diff because you always open files from the www directory.

I did try other solutions but I only got tangled up in path mappings, errors, and debugging problems. All help is appreciated.

**Comment by pjcdawkins** [comment]
Hm. With my method I'm not getting a dialog for function declarations. Just to confirm, it looks a bit like this:

<img src="https://cloud.githubusercontent.com/assets/1465106/6218925/5743d5aa-b61b-11e4-9143-ae414b4b1649.png" width="183" height="207" alt="Screenshot" />

**Comment by jurgenhaas** [comment]
@pjcdawkins that'S correct, declaring the builds directory excluded does solve that one problem but it leaves us with the problem that the debugger is complaining because the debugger doesn't know about the sym-linked directory, it comes from the original build directory.

What would solve the problem is if the --copy option to the build command would work as expected, but it doesn't seem to work - or I mis-read the documentation. I have reported an issue for this at #344
